src/file_storage.cpp

```diff
@@ -564,7 +564,7 @@ namespace libtorrent
 		else
 		{
 			if (m_files.empty())
-				m_name = split_path(path).c_str();
+				m_name = split_path(path);
```

Comment:
split path will return a string with '\0' characters embedded in it. this change will copy the entire string, rather than just the first element. is there a need to copy the entire string?
(and I admit that this is a pretty shitty function interface)